crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -296,15 +296,31 @@ pub struct EncryptionInfo {
     pub verification_state: VerificationState,
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum EncryptionState {
+    Unencrypted,
+    Decrypted(EncryptionInfo),
+    Utd(UtdReason),
```

Comment:
We should probably use the existing `UnableToDecryptInfo` here (and add a "reason" to it).